libr/core/cmd_api.c

```diff
@@ -355,6 +353,174 @@ R_API RCmdStatus r_cmd_call_parsed_args(RCmd *cmd, RCmdParsedArgs *args) {
 	return res;
 }
 
+static size_t strlen0(const char *s) {
+	return s? strlen (s): 0;
+}
+
+static void fill_usage_strbuf(RStrBuf *sb, RCmdDesc *cd, bool use_color) {
+	RCons *cons = r_cons_singleton ();
+	const char *pal_label_color = use_color? cons->context->pal.label: "",
+		   *pal_args_color = use_color? cons->context->pal.args: "",
+		   *pal_help_color = use_color? cons->context->pal.help: "",
+		   *pal_reset = use_color? cons->context->pal.reset: "";
+
+	r_strbuf_appendf (sb, "%sUsage:%s ", pal_label_color, pal_reset);
+	if (cd->help.usage) {
+		r_strbuf_appendf (sb, "%s%s%s", cd->help.usage, pal_args_color, pal_reset);
+	} else {
+		r_strbuf_appendf (sb, "%s %s%s%s", cd->name, pal_args_color, cd->help.args_str, pal_reset);
+	}
+	if (cd->help.group_summary) {
+		r_strbuf_appendf (sb, "   %s# %s%s\n", pal_help_color, cd->help.group_summary, pal_reset);
+	} else {
+		r_strbuf_appendf (sb, "   %s# %s%s\n", pal_help_color, cd->help.summary, pal_reset);
+	}
+}
+
+static size_t update_max_len(RCmdDesc *cd, size_t max_len) {
+	size_t name_len = strlen (cd->name);
+	size_t args_len = strlen0 (cd->help.args_str);
+	if (name_len + args_len > max_len) {
+		return name_len + args_len;
+	}
+	return max_len;
+}
+
+static void print_child_help(RStrBuf *sb, RCmdDesc *cd, size_t max_len, bool use_color) {
+	size_t str_len = strlen (cd->name) + strlen0 (cd->help.args_str);
+	size_t padding = str_len < max_len? max_len - str_len: 0;
+	const char *cd_args_str = cd->help.args_str? cd->help.args_str: "";
+	const char *cd_summary = cd->help.summary? cd->help.summary: "";
+
+	RCons *cons = r_cons_singleton ();
+	const char *pal_args_color = use_color? cons->context->pal.args: "",
+		   *pal_help_color = use_color? cons->context->pal.help: "",
+		   *pal_reset = use_color? cons->context->pal.reset: "";
+	r_strbuf_appendf (sb, "| %s %s%s %*s%s# %s%s\n", cd->name, pal_args_color, cd_args_str, padding, "", pal_help_color, cd_summary, pal_reset);
+}
+
+static char *inner_get_help(RCmd *cmd, RCmdDesc *cd, bool use_color) {
+	if (!cd->help.args_str || !cd->help.summary) {
```

Comment:
Why not `r_return_if_fail()`?

Reply:
I did not want to enforce help for all commands right now as in this transition phase some stuff may be left out and added at some point later.